transformer/classes/dockerfile/pythondockerfilegenerator.go

```diff
@@ -0,0 +1,229 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package dockerfile
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/konveyor/move2kube/common"
+	"github.com/konveyor/move2kube/environment"
+	"github.com/konveyor/move2kube/qaengine"
+	"github.com/konveyor/move2kube/types/qaengine/commonqa"
+	transformertypes "github.com/konveyor/move2kube/types/transformer"
+	"github.com/konveyor/move2kube/types/transformer/artifacts"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	pythonExt           = ".py"
+	pythonMain1         = "__name__ == '__main__'"
+	pythonMain2         = "__name__ == \"__main__\""
+	django              = "django"
+	requirementsTxtFile = "requirements.txt"
+	//DetectedPortsConfigType contains the detected ports
+	DetectedPortsConfigType transformertypes.ConfigType = "DetectedPortsConfigType"
+	//RequirementsTxtPathType points to the requirements.txt file if it's present
+	RequirementsTxtPathType transformertypes.PathType = "RequirementsTxtPathType"
+	// DjangoProjectConfigType is set to true for Django projects
+	DjangoProjectConfigType transformertypes.ConfigType = "DjangoProjectConfigType"
+	// MainPythonFilesPathType points to the .py file path which contains main function
+	MainPythonFilesPathType transformertypes.PathType = "MainPythonFilesPathType"
+)
+
+// PythonDockerfileGenerator implements the Transformer interface
+type PythonDockerfileGenerator struct {
+	Config transformertypes.Transformer
+	Env    *environment.Environment
+}
+
+// PythonTemplateConfig implements python config interface
+type PythonTemplateConfig struct {
+	AppName           string
+	Port              int32
+	MainScriptRelPath string
+	RequirementsTxt   string
+	IsDjangoProject   bool
+}
+
+// Init Initializes the transformer
+func (t *PythonDockerfileGenerator) Init(tc transformertypes.Transformer, env *environment.Environment) (err error) {
+	t.Config = tc
+	t.Env = env
+	return nil
+}
+
+// GetConfig returns the transformer config
+func (t *PythonDockerfileGenerator) GetConfig() (transformertypes.Transformer, *environment.Environment) {
+	return t.Config, t.Env
+}
+
+// findMainScripts returns the path of .py files having the main function
+func findMainScripts(pythonFilesPath []string) ([]string, error) {
+	if len(pythonFilesPath) == 0 {
+		return []string{}, nil
+	}
+	var pythonMainFiles []string
+	for _, pythonFilePath := range pythonFilesPath {
+		pythonFile, err := ioutil.ReadFile(pythonFilePath)
+		if err != nil {
+			logrus.Warnf("Error in reading python file %s: %s.", pythonFilePath, err)
+		} else {
+			if strings.Contains(string(pythonFile), pythonMain1) || strings.Contains(string(pythonFile), pythonMain2) {
+				pythonMainFiles = append(pythonMainFiles, pythonFilePath)
+			}
```

Comment:
@Akash-Nayak This is better done as a regex than contains. My previous comment on contains was only for django search in requirements file. 

For main check, it is important to use regex, because we also need to ensure it is at start of a line, and is more than one possibility for string and spaces. So a regex like say `^__name__\s*==\s*['"]__main__['"]\s*$ might capture most possibilities in a single check.